lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -38,6 +38,27 @@ class Map
   end
 end
 
+module Google
+  module Auth
+    # Extract project_id in initialize.
+    class ServiceAccountCredentials
+      singleton_class.send(:alias_method, :super_make_creds, :make_creds)
+      def self.make_creds(options = {})
+        json_key_io, scope = options.values_at(:json_key_io, :scope)
+        if json_key_io
+          json_key = MultiJson.load(json_key_io.read)
+          project_id = json_key['project_id']
```

Comment:
Do we need to check if `project_id` exists first? Just in case some corrupted credential files are passed in. It would be nice to log a warning and clear instructions to fix it instead of crashing the agent.

Reply:
Ruby hashes return `nil` if the key is not present.

Reply:
Yeah, Ruby does that. Yet I meant something that explicitly indicates the json key is corrupted when `project_id` is not present. Better to error out loudly, right? Kind of similar to what googleauth does [here](https://github.com/google/google-auth-library-ruby/blob/master/lib/googleauth/service_account.rb#L74):

```
        json_key = MultiJson.load(json_key_io.read)
        raise 'missing client_email' unless json_key.key?('client_email')
        raise 'missing private_key' unless json_key.key?('private_key')
```

Reply:
The `project_id` field is optional, so we probably don't want to error out. Setting it to `nil` will default to extracting it out of the email address, which is what we want.